modules/apps/static/portal-configuration/portal-configuration-metatype-api/src/main/java/com/liferay/portal/configuration/metatype/annotations/ExtendedObjectClassDefinition.java

```diff
@@ -77,6 +77,16 @@ public enum Scope {
 			"portletInstanceKey", "portletInstanceId", "portlet-instance"),
 		SYSTEM(null, null, "system");
 
+		public static Scope getScope(String value) {
+			for (Scope scope : values()) {
+				if (scope._value.equals(value)) {
+					return scope;
+				}
+			}
+
+			throw new IllegalArgumentException("Invalid value " + value);
+		}
+
```

Comment:
(optional) Sorry about this. I assumed the scope name matched the enum name. Not sure if Brian will like this, but you can keep it if you want.

Reply:
I don't think it's a big change, and I find it useful. Let's see if Brian likes it too.